### PR TITLE
UnixPB: Add Fedora 40 Support

### LINKED ISSUE #3599: Ansible request for Fedora playbook corrections
Please put the name of the software product (and affected platforms if relevant) in the title of this issue

Delete as appropriate from this list:

- Bug in ansible playbook

Details:

- libnss3.so is not a valid package on Fedora so should be removed (From [this thread](https://adoptium.slack.com/archives/C53GHCXL4/p1697552390431979) I thought it had been but seems not universally). We've now removed it from the [UBI8 dockerfile](https://github.com/adoptium/infrastructure/pull/3595/files) but it is still in the playbooks.
- Fedora (verified on F37/riscv64) also seems to be trying to install java 1.7.0 which is not in its repositories any more so that should also be removed.
- `Create Japanese locale` in the `Common` role doesn't seem to be working on F37/rv64: `/etc/locale.gen and /var/lib/locales/supported.d/local are missing. Is the package \"locales\" installed?`
- Surprisingly `tar` is missing from the machines by default so the `ant-contrib` task falls over failing to extract that tarball
- Also while not generic to Fedora the gcc tags (`gcc_48,gcc_7,gcc_9,gcc_10,gcc_11`) need to be expliclty skipped on riscv64 otherwise it tries to download and install the adoptopenjdk gcc versions which aren't available for riscv64.
- Nagios_plugins role does not have a Fedora configuration so that role fails and `nagios_plugins` had to be added to the skip list in the previous bullet point (EDIT: That doesn't work as it wipes out the vendor files retrieval so the jenkins role fails - copied out that tag from the vendor files role for now)

Found as part of testing the machine referenced in https://github.com/adoptium/infrastructure/issues/3598#issuecomment-2167899948

Noting that:
- The [VPC runs with Fedora 35](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/OS=Fedora35,label=vagrant/1877/console) do not appear to be falling foul of this
- VPC should probably be updated to use Fedora 40 now.

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3599 

Correct Unix playbooks to allow execution on Fedora 40.

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

Fedora 40: VPC : https://ci.adoptium.net/job/VagrantPlaybookCheck/1974/ 
Ubuntu 22: VPC : https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Ubuntu2204,label=vagrant/1975/
Debian 10: VPC : https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Debian10,label=vagrant/1975/

### CODE REVIEW

File: ansible/pbTestScripts/vmDestroy.sh [hunk 1]
```diff
@@ -61,6 +61,9 @@ checkOS() {
                         osToDestroy="D8" ;;
                 "Debian10" | "debian10" | "D10" | "d10" )
                         osToDestroy="D10" ;;
+								"Fedora40" | "fedora40" | "F40" | "F40" )
```
Comment (karianna) [inline ansible/pbTestScripts/vmDestroy.sh:64]: whitespace seems out

**Review by karianna — CHANGES_REQUESTED** [review]

File: ansible/pbTestScripts/vmDestroy.sh (hunk 1 above)
Comment (steelhead31) [reply ansible/pbTestScripts/vmDestroy.sh]: fixed, thanks!.

**Review by karianna — APPROVED** [review]

**Review by sxa — APPROVED** [review]
I'm ok with this, but noting:
- are there any implications on older Fedora versions of completely removing `libnss3.so` here - I think that was added for some of the security tests.
- Noting that in the `nasm` playbook we have a mix of syntax: `ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "22")` vs `(ansible_distribution == "Ubuntu" and ansible_distribution_major_version | int >= 24)`. I definitely have a recollection - possibly ansible version specific - that one is better than the other so maybe we should standardise. I have a feeling it was the `| int` version that was preferable but don't hold me to that :-)

**Comment by steelhead31** [comment]
> *   are there any implications on older Fedora versions of completely removing `libnss3.so` here - I think that was added for some of the security tests.

For the currently supported versions of Fedora ( 39 & 40 ) the libnss3 libraries are installed via nss-devel & nss-util packages, so removing it has no impact on the currently supported versions.  
  
The vagrant playbook checks run the SSL tests as part of the TEST\_JDK flag, so this would highlight any issues for the Fedora 40 run.  
 

> *   Noting that in the `nasm` playbook we have a mix of syntax: `ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "22")` vs `(ansible_distribution == "Ubuntu" and ansible_distribution_major_version | int >= 24`

I've standardised this to be consistent for all the major version checks. I believe the int comparison is a more reliable comparison for numeric values, particularly as using \<= with string operations is a bit inconsistent. the string operations are fine for a simple ==.

**Comment by sxa** [comment]
@steelhead31 I guess this supercedes https://github.com/adoptium/infrastructure/pull/3663?

**Comment by steelhead31** [comment]
> @steelhead31 I guess this supercedes #3663?

Yup, I hadnt seen the earlier one.

**Pull request merged**